src/vit-based/vit.jl

```diff
@@ -0,0 +1,174 @@
+include("utilities.jl")
+
+# Utility function for applying LayerNorm before a block
+prenorm(planes, fn) = Chain(fn, LayerNorm(planes))
+
+"""
+      mlpblock(inplanes, hidden_planes, dropout=0.)
+  
+MLP block as used in the base ViT architecture.
+([reference](https://arxiv.org/abs/2010.11929)).
+
+# Arguments
+- `inplanes`: number of input channels
+- `hidden_planes`: number of hidden channels
+- `dropout`: dropout rate
+"""
+function mlpblock(inplanes, hidden_planes, dropout = 0.)
+  Chain(Dense(inplanes, hidden_planes, gelu),
+        Dropout(dropout),
+        Dense(hidden_planes, inplanes),
+        Dropout(dropout))
+end
+
+struct attention
+  heads
+  scale
+  qkvlayer
+  outlayer
+end
```

Comment:
Struct names should be capitalized in Julia by convention. But I'd suggest a similar redesign here as I did with MLPMixer. A lot of the functions, like `chunk` or `batchedmul` are trying to replicate operations across various heads. This is more easily expressed using `Parallel` without needing a custom forward pass.

Reply:
The problem is that `chunk` is not exactly across the heads so much as it is to split `q`, `k`, and `v` from each other. And while the subsequent reshaping operation is definitely something that can be done in parallel, the multiplication after that isn't symmetric - https://github.com/chengchingwen/Transformers.jl/blob/master/src/basic/mh_atten.jl, for example, is another place where the implementation is similar

Reply:
True, and if this is part of Flux, then it is worth introducing a new type. I'll post a more complete suggestion later today.